tests/tests-hz38/src/main/java/com/hazelcast/simulator/tests/cardinalityestimator/CardinalityEstimatorTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.simulator.tests.cardinalityestimator;
+
+import com.hazelcast.cardinality.CardinalityEstimator;
+import com.hazelcast.core.IAtomicLong;
+import com.hazelcast.simulator.test.AbstractTest;
+import com.hazelcast.simulator.test.annotations.Setup;
+import com.hazelcast.simulator.test.annotations.Teardown;
+import com.hazelcast.simulator.test.annotations.TimeStep;
+import com.hazelcast.simulator.test.annotations.Verify;
+
+import java.util.Random;
+
+import static org.junit.Assert.assertTrue;
+
+public class CardinalityEstimatorTest extends AbstractTest {
+
+    // properties‚
+    public long batchSize = 1000000;
+    public double tolerancePercentage = 0.1;
+    public double duplicateKeysPercentage = 0.1;
+
+
+    private IAtomicLong elementCounter;
+    private IAtomicLong rangeSelectorCounter;
+    private CardinalityEstimator cardinalityEstimator;
+    private final Random random = new Random();
+
+    @Setup
+    public void setup() {
+        cardinalityEstimator = targetInstance.getCardinalityEstimator(name);
+        elementCounter = targetInstance.getAtomicLong(name);
+        rangeSelectorCounter = targetInstance.getAtomicLong("rangeSelector");
+        rangeSelectorCounter.set(100L);
+    }
+
+    @TimeStep
+    public void loadData() {
+        Range batchRange = getRange();
+        logger.info("Running range : " + batchRange.toString());
+        for (long i = batchRange.startIndex; i < batchRange.endIndex; i++) {
+            cardinalityEstimator.add(i);
+        }
+    }
+
+    @Verify
+    public void verify() {
+        int tolerance = (int) (elementCounter.get() * tolerancePercentage);
+        assertTrue(Math.abs(cardinalityEstimator.estimate() - elementCounter.get()) < tolerance);
+    }
+
+    private Range getRange() {
+        if (rangeSelectorCounter.get() % (duplicateKeysPercentage * 100) != 0) {
```

Comment:
Since rangeSelectorCount is never going to change, it is a constant. So the choice is going to be constant as well. For the time being, let the 'getRange' just return 'nextRange'. Ignore the random part for now.